### PR TITLE
GTEST/STREAM: add more tests + bugfix

### PR DESCRIPTION
**Pull request by evgeny-leksikov**

### CODE REVIEW

**Comment by evgeny-leksikov** [comment]
bot:ornl:retest

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5266/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5270/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@MattBBaker pls post ORNL failure

**Comment by MattBBaker** [comment]
unsigned comparison.
test/gtest/ucp/test_ucp_stream.cc line 195
```
  CXX    ucp/gtest-test_ucp_peer_failure.o
cc1plus: warnings being treated as errors
In file included from /var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/common/test_helpers.h:11,
                 from /var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/common/test.h:10,
                 from /var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/ucp/ucp_test.h:17,
                 from /var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/ucp/test_ucp_stream.cc:10:
/var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/common/gtest.h: In function ‘testing::AssertionResult testing::internal::CmpHelperGT(const char*, const char*, const T1&, const T2&) [with T1 = size_t, T2 = int]’:
/var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/ucp/test_ucp_stream.cc:195:   instantiated from here
/var/lib/jenkins/jobs/UCX/workspace/contrib/../test/gtest/common/gtest.h:18999: error: comparison between signed and unsigned integer expressions
make[2]: *** [ucp/gtest-test_ucp_stream.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[2]: Leaving directory `/var/lib/jenkins/jobs/UCX/workspace/build-tests/1511867662/test/gtest'
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory `/var/lib/jenkins/jobs/UCX/workspace/build-tests/1511867662'
make: *** [all] Error 2
```

**Comment by mellanox-github** [comment]
Test FAILed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5284/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
bot:mlx:retest

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5286/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@yosefe pls take a look

File: src/ucp/stream/stream_recv.c [hunk 1]
```diff
@@ -304,7 +304,7 @@ ucp_stream_am_handler(void *am_arg, void *am_data, size_t am_length,
 
     ep_stream = ep->ext.stream;
 
-    if (ep->flags & UCP_EP_FLAG_STREAM_IS_QUEUED) {
+    if (ucp_stream_ep_is_queued(ep)) {
```
Comment (yosefe) [inline src/ucp/stream/stream_recv.c]: need to do ucs_queue_push(&ep_stream->data, &rdesc->stream_queue) before this check to avoid code duplication

File: test/gtest/ucp/test_ucp_stream.cc [hunk 2]
```diff
@@ -17,6 +20,40 @@ class test_ucp_stream : public ucp_test {
         return params;
     }
 
+    virtual void init() = 0;
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_stream.cc]: why is this needed?

File: test/gtest/ucp/test_ucp_stream.cc [hunk 3]
```diff
@@ -73,10 +103,113 @@ void test_ucp_stream::do_send_recv_data_test(ucp_datatype_t datatype)
...
 
     EXPECT_EQ(roffset, ssize);
-    sbuf.resize(ssize, 's');
-    EXPECT_EQ(sbuf, rbuf);
+    EXPECT_EQ(check_pattern, rbuf);
 }
 
+void test_ucp_stream::do_send_recv_test(ucp_datatype_t datatype)
+{
+    std::vector<char> sbuf(size_t(16)*1024*1024, 's');
```
Comment (yosefe) [inline test/gtest/ucp/test_ucp_stream.cc]: why need to cast size_t?
also code style - space around operators

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/5288/ for details (Mellanox internal link).

**Comment by evgeny-leksikov** [comment]
@yosefe good to go?
